opengrok-indexer/src/main/java/org/opengrok/indexer/search/SearchEngine.java

```diff
@@ -205,18 +205,17 @@ private void searchMultiDatabase(List<Project> projectList, boolean paging) thro
     }
 
     private void searchIndex(IndexSearcher searcher, boolean paging) throws IOException {
-        collector = TopScoreDocCollector.create(hitsPerPage * cachePages, Short.MAX_VALUE);
+        collectorManager = new TopScoreDocCollectorManager(hitsPerPage * cachePages, Short.MAX_VALUE);
         Statistics stat = new Statistics();
-        searcher.search(query, collector);
-        totalHits = collector.getTotalHits();
+        hits = searcher.search(query, collectorManager).scoreDocs;
+        totalHits = searcher.count(query);
```

Comment:
Formerly `totalhits` would have been bounded by the `Short.MAX_VALUE` value of `totalHitsThreshold` (on the `TopScoreDocCollector`). For very large indexes, is there any possible performance penalty getting the full, "fully accurate" `int` value of `totalHits`?